### DISCUSSION TITLE
Mac Mini M1 "zsh: command not found"

### DISCUSSION THREAD

**Discussion by ryadaj** [Homebrew/discussions#1719] [Getting started] [answered]
Just upgraded to Mac Mini M1, previous was a 2017 MBP i5, on which I did not have this issue.

Tried to install Homebrew, but at the end of installation when I typed "brew help" I got: "zsh: command not found: brew" This fixed it:

export PATH="/opt/homebrew/bin:$PATH"

But then when I installed espanso and typed "espanso edit" I got "zsh: command not found: espanso". This fixed it again:

export PATH="/opt/homebrew/bin:$PATH"

But then when I closed and opened Terminal, both the above problems exist.

Espanso is functioning fine, despite these issues, and any time I update espanso with new text expansions, all I have to do to update the config is insert the above and then restart espanso, but it would be nice not to have to insert the above command every time.

Any ideas would be appreciated!

**Comment by carlocab** [comment, marked as answer]
Did you follow the instructions shown to you by the installer when it was done? It should have shown you something like
```
- Add Homebrew to your PATH in ~/.zprofile:
    echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
    eval $(/opt/homebrew/bin/brew shellenv)
```

**Reply by ryadaj** [reply]
I did yeah. And I'm certain I did because it was my second time around.

Do you reckon reinstalling Homebrew would fix it? If so, how would I go about doing that? Not really a tech savvy person but I can figure things out.

**Reply by carlocab** [reply]
I'm not sure reinstalling Homebrew would fix this. What exactly is espanso? It doesn't seem to come from any formula or cask we ship. How did you install it?

**Reply by ryadaj** [reply]
Espanso is a text expander 😄  I installed it using homebrew https://espanso.org/

**Reply by carlocab** [reply]
Espanso comes from https://github.com/federico-terzi/homebrew-espanso. We don't maintain this repository, so you'll have to ask there about your issues with espanso.

What is the output of `brew config` and `brew doctor`? [Please don't post a screenshot of the output](https://meta.stackoverflow.com/a/285557). Instead, paste the text directly here. You can wrap it in \`\`\` in the line before and the line after to format it properly.

**Reply by ryadaj** [reply]
Here is what I get

```
ryan@Ryans-Mac-mini ~ % brew config
zsh: command not found: brew
ryan@Ryans-Mac-mini ~ % brew doctor
zsh: command not found: brew
```

Here is what I get for ```brew config``` when first entering: ```export PATH="/opt/homebrew/bin:$PATH"```

```
ryan@Ryans-Mac-mini ~ % brew config
HOMEBREW_VERSION: 3.1.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 33e71c99dc40c0a134a236fffe2594ae6572d83c
Last commit: 12 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c39245de735ce10a9a9d53c09db84bb12ab92a81
Core tap last commit: 5 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.4-arm64
CLT: 12.5.0.0.1.1617976050
Xcode: N/A
Rosetta 2: false
```

And for ```brew doctor```

```
ryan@Ryans-Mac-mini ~ % brew doctor
Your system is ready to brew.
```

**Reply by carlocab** [reply]
You need to do this:
```
- Add Homebrew to your PATH in ~/.zprofile:
    echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
    eval $(/opt/homebrew/bin/brew shellenv)
```
Please follow the suggested commands carefully.

**Reply by ryadaj** [reply]
Yup that worked. Thank you so much! 🙇‍♂️ 🙏

**Reply by natnaelhhaile** [reply]
I am sorry for piggybacking on an old discussion, but I've got the same issue. Only difference is, it works after I added the path as you specified above, but I kinda have to do it every time I reboot my terminal. I am using Macbook air m2. I even checked my ~/.zprofile and the 'eval $(/opt/homebrew/bin/brew shellenv)' line is there. What could I be missing?

**Comment by gromgit** [comment]
Open a new Terminal, then type the following commands:
```
env | grep -i homebrew
tail ~/.zprofile
```
What's the output of the above commands?

**Reply by ryadaj** [reply]
![Screen Shot 2021-06-22 at 09 19 03](https://user-images.githubusercontent.com/82765301/122931440-e47ea100-d33a-11eb-8443-29c7e6496d82.png)

**Reply by carlocab** [reply]
It looks like `~/.zprofile` does not exist, which means you didn't follow the instructions referenced in https://github.com/Homebrew/discussions/discussions/1719#discussioncomment-899843. Running those commands will have created a `~/.zprofile`.

**Reply by ryadaj** [reply]
Hmm looks like I messed up. Will reply to you above.
